heron/api/src/scala/com/twitter/heron/streamlet/scala/impl/StreamletImpl.scala

```diff
@@ -0,0 +1,291 @@
+//  Copyright 2018 Twitter. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+package com.twitter.heron.streamlet.scala.impl
+
+import com.twitter.heron.streamlet.{KeyValue, KeyedWindow}
+import com.twitter.heron.streamlet.{
+  JoinType,
+  SerializableTransformer,
+  WindowConfig
+}
+import com.twitter.heron.streamlet.scala.{Sink, Streamlet}
+import com.twitter.heron.streamlet.scala.converter.ScalaToJavaConverter._
+
+object StreamletImpl {
+
+  def toScalaStreamlet[R](
+      javaStreamlet: com.twitter.heron.streamlet.Streamlet[R]): Streamlet[R] =
+    new StreamletImpl[R](javaStreamlet)
```

Comment:
what does this imply when we pass the javaStreamlet as an arg and have it listed with a colon, the notation is a bit confusing here, can you add some comments

Reply:
After related `user defined function` is applied(e.g: `map`), it will return `Java Streamlet` object type and this needs to be transformed to `Scala Streamlet `object type again before exposing the user so in order to achieve this, `toScalaStreamlet` function accept `Java Streamlet` Interface and transforms it to `Scala Streamlet` Trait type.